index.js

```diff
@@ -873,6 +878,49 @@ class Batch {
   }
 }
 
+class EntryWatcher extends ReadyResource {
+  constructor (bee, key) {
+    super()
+
+    bee._watchers.add(this)
```

Comment:
is this not based on current main?

Reply:
use a different set also, cleaner and not polymorphic

Reply:
I branched off before the arrayset change was merged in. Fixed no, and also using a different set.

Note: both sets are currently triggered by the same onappend/ontruncate events. I think that's fine though, no more polymorphism involved (the Hyperbee's onappend just iterates both sets of watchers)